docs: add a windows know issue (#42028)
angular/angular: docs/BAZEL.md

 (error: 2)
ERROR: no such package '@npm//@bazel/protractor': mkdir -p _ failed: 
STDOUT:

STDERR:
java.io.IOException: ERROR: src/main/native/windows/process.cc(202): CreateProcessW("mkdir" -p _): The system cannot find the file specified.     
 (error: 2)
child_process.js:655
    throw err;
    ^

Error: Command failed: yarn --cwd "D:\OpenSource\angular" --silent bazel info bazel-bin
    at checkExecSyncError (child_process.js:616:11)
    at execSync (child_process.js:652:15)
    at exec (D:\OpenSource\angular\scripts\build\package-builder.js:132:18)
    at Object. (D:\OpenSource\angular\scripts\build\package-builder.js:40:18)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18) {
  status: 1,
  signal: null,
  output: [ null, Buffer(0) [Uint8Array] [], null ],
  pid: 15276,
  stdout: Buffer(0) [Uint8Array] [],
  stderr: null
}
```

> The `msys64` library is required to build angular. Make sure you have `C:\msys64\usr\bin` in your system PATH rather than the user `PATH`. After that, a `git clean -xfd`, `yarn`, and `node scripts\build\build-packages-dist.js` will resolve this issue. 

Reviewer: ```suggestion
The `msys64` library and associated tools (like `mkdir`) are required to build Angular.

Make sure you have `C:\msys64\usr\bin` in the "system" `PATH` rather than the "user" `PATH`. 

After that, a `git clean -xfd`, `yarn`, and `node scripts\build\build-packages-dist.js` should resolve this issue. 
```

Author: Sure. thanks for the copy-text.
